.circleci/config.yml

```diff
@@ -30,11 +31,20 @@ jobs:
           echo "$WEBAPP_GIT_COMMIT"
 
           FILE_DIST=dist.tar.gz
-          until curl --max-time 5 -f -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; do echo waiting for webapp build; sleep 110; done;
-          if [ ! -f $FILE_DIST ]; then
-            npm ci && make build
-          else
+          curl --version
+
+          runtime="2 minute"
+          endtime=$(date -ud "$runtime" +%s)
+          while [[ $(date -u +%s) -le $endtime ]]; do
+              echo "Waiting for webapp git commit $WEBAPP_GIT_COMMIT with sleep 5: `date +%H:%M:%S`"
+              curl --max-time 30 -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/$WEBAPP_GIT_COMMIT/mattermost-webapp.tar.gz
```

Comment:
```suggestion
              curl --max-time 30 -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/$WEBAPP_GIT_COMMIT/mattermost-webapp.tar.gz && break
```

Reply:
I think we can exit the loop on success otherwise in most case we end up downloading multiple times and unnecessarily waiting the whole duration.